src/cluster.c

```diff
@@ -3963,6 +4021,8 @@ int clusterAddSlot(clusterNode *n, int slot) {
  * Returns C_OK if the slot was assigned, otherwise if the slot was
  * already unassigned C_ERR is returned. */
 int clusterDelSlot(int slot) {
+    /* Cleanup the channels as part of slot deletion. */
+    removeChannelsInSlot(slot);
```

Comment:
I know we talked about this, but it looks like this is called generally anytime a slot is deleted, even if it's not one we own.

Reply:
Added a condition to verify if the node is the owner of the slot and then cleanup.